### ISSUE TITLE
camelcase rule false report with optional chaining property

### ISSUE THREAD

**Issue by diondirza** [eslint/eslint#10848]
**Tell us about your environment**

* **ESLint Version:** 5.5.0
* **Node Version:** 8.11.4
* **npm Version:** 6.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
`camelcase: ['error', { properties: 'never' }],`
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const dataSubCat = await graphql(
    CategoryQuery,
    { id: dataCat?.categories[0]?.cat_id || -1 },
    mapResult,
    'Category Sub query',
);
```

```bash
yarn lint
```

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
<img width="814" alt="screen shot 2018-09-10 at 10 56 14" src="https://user-images.githubusercontent.com/10954870/45275941-c88f1500-b4e9-11e8-81d6-417dfde4b875.png">

**Comment by villesau** [comment]
I would be happy to fix this, but I don't know how to set proper parser in tests that would support optional chaining.

**Comment by ilyavolodin** [comment]
It sounds like we are not checking exceptions against `OptionalMemberExpression` node. However, that's node is still not part of the ECMAScript, so the modification for this rule would need to be done in `eslint-plugin-babel`.

**Comment by diondirza** [comment]
I see, I thought I should put the issue here, anyway thanks for the feedback, will follow up this issue on proper repo then

**Comment by tsheaff** [comment]
@diondirza @Aladdin-ADD any progress on this? if not we can jump in and make a PR if it helps @ajones

**Comment by diondirza** [comment]
I think @villesau already had a PR to fix this.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in this issue. However, I'm closing this for the same reasoning as described in https://github.com/eslint/eslint/issues/10848#issuecomment-422488992 and https://github.com/eslint/eslint/pull/10868#issuecomment-426108973. I think it was an oversight that this wasn't originally closed at the same time as https://github.com/eslint/eslint/pull/10868.

**Comment by villesau** [comment]
FYI: eslint-plugin-babel now supports optional chaining: https://github.com/babel/eslint-plugin-babel/pull/163
